editor/src/components/canvas/controls/selection-area-helpers.ts

```diff
@@ -0,0 +1,145 @@
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { mapDropNulls } from '../../../core/shared/array-utils'
+import * as EP from '../../../core/shared/element-path'
+import { ElementPathTrees } from '../../../core/shared/element-path-tree'
+import {
+  ElementInstanceMetadata,
+  ElementInstanceMetadataMap,
+} from '../../../core/shared/element-template'
+import {
+  CanvasPoint,
+  CanvasRectangle,
+  WindowPoint,
+  canvasRectangle,
+  isFiniteRectangle,
+  rectContainsPoint,
+  rectangleContainsRectangle,
+  rectanglesOverlap,
+} from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+
+export function getPossibleElementsUnderMouse(
+  metadata: ElementInstanceMetadataMap,
+  pathTrees: ElementPathTrees,
+  selectedViews: ElementPath[],
+): ElementInstanceMetadata[] {
+  const selectableElements = mapDropNulls((path) => {
+    return MetadataUtils.findElementByElementPath(metadata, path)
+  }, MetadataUtils.getAllCanvasSelectablePathsOrdered(metadata, pathTrees))
+
+  const nonSelectableElementsPossiblyUnderMouse = Object.values(metadata).filter((e) =>
+    selectableElements.some((other) => EP.isDescendantOf(e.elementPath, other.elementPath)),
+  )
+
+  return [
+    ...selectableElements,
+    ...nonSelectableElementsPossiblyUnderMouse,
+    ...mapDropNulls(
+      (path) => MetadataUtils.findElementByElementPath(metadata, path),
+      selectedViews,
+    ),
+  ]
+}
+
+export const getElementsUnderSelectionArea = (
+  metadata: ElementInstanceMetadataMap,
+  selectionAreaCanvasRect: CanvasRectangle,
+): ElementPath[] => {
```

Comment:
We already have functions to select elements under a point in `dom-lookup.ts`, e.g. `getAllTargetsAtPoint` which uses dom lookup or `getAllTargetsAtPointAABB` which uses the global frames of elements.

I think those could be reused, partially to avoid duplication, partially because they include all kinds of logic which is necessary to select the correct elements. E.g. as far as I see with this PR you can select locked and even hidden elements, also elements which are behind other elements in z-order (or is that deliberate?)

Reply:
I believe this change has made it possible to highlight and select and element whilst 1px to the left of it